.github/workflows/deploy-treasury-factory-contract.yml

```diff
@@ -2,6 +2,13 @@ name: Deploy Treasury Factory contract
 
 on:
   workflow_dispatch: # This allows manual triggering
+  push:
```

Comment:
Do we want to deploy everytime?

Reply:
Only when the specified paths have changes.